### DISCUSSION TITLE
0.4.0

### DISCUSSION THREAD

**Discussion by mcndt** [mcndt/obsidian-toggl-integration#38] [Releases]
This is the biggest update so far! This update enables you to visualize your Toggl time entry data right inside your Obsidian notes.

To learn how, check out the [getting started](https://github.com/mcndt/obsidian-toggl-integration/wiki/Creating-Toggl-reports-inside-notes) page!

### ✨ Features

- You can now generate Toggl reports inside notes using code blocks with the language '`toggl`'
- Generate summary reports
- Generate list reports

### 🐛 Fixes

- (Probably) fixed [a bug](https://github.com/mcndt/obsidian-toggl-integration/issues/34) where the sidebar periodically clears due to failing API requests.

<hr /><em>This discussion was created from the release <a href='https://github.com/mcndt/obsidian-toggl-integration/releases/tag/0.4.0'>0.4.0</a>.</em>

**Comment by tintinwulia** [comment]
This is an amazing feature! Thank you so much @mcndt! 
I also love it that your documentation is so clear with great examples that are easy to understand for a beginner like me. 

It would be fantastic though if the codeblock would also parse Templater syntax. Currently I keep getting the error message '"<%" is not a keyword, ISO-formatted date, decimal number or string."' What I'm trying to do is to have it inserted automatically in my Daily Note template.

**Reply by mcndt** [reply]
Hi Tintinwulia,

I like that you're trying to integrate with Templater. That's something I've been meaning to get working myself but haven't yet made the time for.

Would you mind sharing the template that you're trying to create? That way I could troubleshoot the problem better.

**Reply by tintinwulia** [reply]
Hi @mcndt, 

The context: I'm using Periodic Notes to create a Daily Note every day. When I click today's date on the Calendar, Periodic Notes would create a Daily Note according to a template that I've setup, with Templater installed. With Templater's syntax, this template automatically put today's date as the Daily Note's title, and several other things as the content (headings) of the note. I want to have a summary/list of my Toggl stats for the day in each Daily Note, so this is what I tried:
```toggl
LIST <% tp.date.now(format: string = "YYYY-MM-DD")%>
GROUP BY PROJECT
SORT DESC
```
Again - I'm a beginner, so I'm not really sure whether I'm doing this correctly. 

Thank you!

**Reply by mcndt** [reply]
This works for me:

````
```toggl
LIST FROM <% tp.date.now("YYYY-MM-DD") %> TO <% tp.date.now("YYYY-MM-DD") %>
GROUP BY PROJECT
SORT DESC
```
````

**Reply by tintinwulia** [reply]
I copied and pasted your script and still got the same error message. In principle, the parsing stops at "<%".

<img width="607" alt="Screen Shot 2022-08-24 at 5 47 39 pm" src="https://user-images.githubusercontent.com/84353462/186361526-ab812ab3-49fb-4ef6-8e8b-5c7378c0304a.png">

**Reply by mcndt** [reply]
I think you misunderstand, the `<% ... %>` syntax is a template. You then have to render the template using the Templater plugin. For example, you can use the Templater command "Replace templates in the active file".

![image](https://user-images.githubusercontent.com/23149353/189441296-42847247-4c50-45c6-8ebf-886b6b4957e4.png)
